### ISSUE TITLE
Make seeds real entities

### ISSUE THREAD

**Issue by byorgey** [swarm-game/swarm#502]
Originally proposed by @xsebek .  See the discussion at #490 .

**Comment by byorgey** [comment]
I think I still like this idea.  Here are some more concrete thoughts:
- For each `growable` entity named `X`, we automatically generate another entity named `X seed` with default attributes.
- ...*unless* there is *already* an entity named `X seed`, in which case we just leave it alone.  This way you can define seeds with custom attributes, if you want.
- There probably should be some way to customize how often seeds drop.  For example, some entities might drop a lot of seeds every time they are harvested, where others might only occasionally drop one seed.
- We might want to add a new command `plant` for planting seeds as opposed to `place` which would simply place a seed entity on the ground with no special behavior.  This is analogous to the reason we added `harvest` vs `grab`, so that you can control when you do and don't want growth to happen.

**Comment by xsebek** [comment]
Good idea, you might be placing seeds just to give them to another robot. :+1:

**Comment by xsebek** [comment]
If we order the unions right, then the generated seeds will be overwritten with explicit seeds. :slightly_smiling_face:

**Comment by byorgey** [comment]
Now that we have a `sow` command and more robust growth mechanisms (#1817), I'd like to revisit this.  For a very brief recap of the context here:

- It needs to be possible to simply `place` and `grab` things without causing any growth to happen (see the long discussion at #490 for motivation).
- Growing/farming stuff should be possible for some entities, but with a bit more effort.
- In #490 three alternatives were considered, and we ended up implementing option 2, namely, adding a `harvest` command which acts like `grab` but causes an entity to immediately start regrowing (#503).
    - This seemed like the simplest sensible option, and indeed, I think it probably was the right choice at the time.
    - However, it has some weird issues: in particular, if you repeatedly `place` and then `harvest` on the same cell, you end up with a bunch of different seed robots all growing at the same time.
    - It also seems kind of unrealistic.

I'm still brainstorming here, but concretely, I would propose the following:
- We currently have `yields` which specifies what entity will be obtained from either `grab` or `harvest`.
- I propose generalizing `yields` to optionally contain a more complex record, something like this:
  ```
  yields:
    grab: oak tree
    harvest:
      - products: [(1, log), (1, branch)]
        weight: 2
      - products: [(1, log), (2, branch)]
        weight: 2
      - products: [(1, log), (1, branch), (2, acorn)]
        weight: 1
  ```
  or something like that.  The point is that (1) we can (optionally) separately specify what is yielded when executing `grab` vs `harvest`; (2) we can yield multiple entities, with a weighted random choice among alternatives.  So in this example `harvest`ing a tree only drops an `acorn` 1/5 of the time on average.
- We might also want to make separate `pickable` vs `harvestable` properties.  *e.g.* I could imagine that some entities can be `harvest`ed but not `grab`bed or vice versa.
- Then we can just reuse the new `sow` command for actually planting things.
- Of course we would want to make sure we have a device that enables `sow`, etc.
